src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1685,6 +1685,22 @@ protected override void EnterParameter(ParameterSymbol parameter)
                 if (slot > 0) SetSlotState(slot, true);
                 NoteWrite(parameter, value: null, read: true);
             }
+
+            if (parameter is SourceComplexParameterSymbol { ContainingSymbol: LocalFunctionSymbol or LambdaSymbol, AttributeDeclarationList.Count: > 0 } sourceComplexParam)
```

Comment:
>SourceComplexParameterSymbol [](http://example.com/codeflow?start=29&length=28)

It looks like ```LambdaParameterSymbol```s are not going to pass this test. Are we testing lambdas?

Reply:
Good catch. Added tests for lambdas: 30a4bc90b5a257d56d902d9e096351e77eb2c9c1 and fixed the code: 21033da490c4fdf93bf93d4b20ae76fc511d763f